src/box/memtx_tx.h

```diff
@@ -48,6 +47,56 @@ extern "C" {
  */
 extern bool memtx_tx_manager_use_mvcc_engine;
 
+enum memtx_tx_alloc_type {
+	MEMTX_TX_ALLOC_TRACKER = 0,
+	MEMTX_TX_ALLOC_CONFLICT = 1,
+	MEMTX_TX_ALLOC_MAX = 2,
+};
+
+extern const char *memtx_tx_alloc_type_strs[];
+
+/**
+ * Memtx_tx allocation objects for memtx_tx_region and memtx_tx_mempool.
+ */
+enum memtx_tx_alloc_object {
+	/**
+	 * Object of type struct memtx_tx_conflict.
+	 */
+	MEMTX_TX_OBJECT_CONFLICT = 0,
+	/**
+	 * Object of type struct tx_conflict_tracker.
+	 */
+	MEMTX_TX_OBJECT_CONFLICT_TRACKER = 1,
+	/**
+	 * Object of type struct tx_read_tracker.
+	 */
+	MEMTX_TX_OBJECT_READ_TRACKER = 2,
+	MEMTX_TX_OBJECT_MAX = 3,
+};
+
+/**
+ * Status of story. Describes the reason why it is not deleted.
+ * In the case when story fits several statuses at once, status with
+ * least value is chosen.
+ */
+enum memtx_tx_story_status {
+	/**
+	 * The story is used directly by some transactions.
+	 */
+	MEMTX_TX_STORY_USED = 0,
+	/**
+	 *  The story can be used by a read view.
+	 */
+	MEMTX_TX_STORY_READ_VIEW = 1,
+	/**
+	 * The story is used for gap tracking.
+	 */
+	MEMTX_TX_STORY_TRACK_GAP = 2,
+	MEMTX_TX_STORY_MAX = 3,
+};
+
+extern const char *memtx_tx_story_status_strs[];
```

Comment:
Nit: why not use `ENUM` and friends from src/trivia/util.h?

Reply:
AFAICS, `ENUM` builds strs from enum values' names, but I want them to be different.

Reply:
I meant using in the following way:
https://github.com/tarantool/tarantool/blob/bb3ed32166688be0b1410a2971bc8380a47be74c/src/lib/core/errinj.h#L70-L174

https://github.com/tarantool/tarantool/blob/bb3ed32166688be0b1410a2971bc8380a47be74c/src/lib/core/errinj.c#L43-L47

Instead of an array of `struct errinj` you could declare an array of `char *`.

Reply:
Cool, but I think that this feature is very useful with large enums, but mine is not.